Maven Central: fix deploy pipeline

OSSRH ist not available anymore:

> mvn deploy
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:3.1.4:deploy (default-deploy) on project basex-parent: Failed to deploy artifacts: Could not transfer artifact org.basex:basex-parent:pom:12.3 from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Transfer failed for https://oss.sonatype.org/service/local/staging/deploy/maven2/org/basex/basex-parent/12.3/basex-parent-12.3.pom 402 Payment Required -> [Help 1]

--- a/basex-core/src/main/java/org/basex/query/value/Value.java
+++ b/basex-core/src/main/java/org/basex/query/value/Value.java
@@ -128,7 +128,7 @@ public abstract class Value extends Expr implements Iterable<Item> {
   /**
    * Returns a subsequence with the given start and length.
    * @param pos position of first item (>= 0)
-   * @param length number of items (1 < length < size())
+   * @param length number of items (1 &lt; length &lt; size())
    * @param job interruptible job
    * @return new subsequence
    */

--- a/basex-core/src/main/java/org/basex/query/value/array/XQArray.java
+++ b/basex-core/src/main/java/org/basex/query/value/array/XQArray.java
@@ -131,7 +131,7 @@ public abstract class XQArray extends XQStruct {
   /**
    * Returns a subsequence with the given start and length.
    * @param pos position of first member (>= 0)
-   * @param length number of members (1 < length < size())
+   * @param length number of members (1 &lt; length &lt; size())
    * @param job interruptible job
    * @return new subarray
    */

--- a/pom.xml
+++ b/pom.xml
@@ -222,16 +222,12 @@
 
   <profiles>
     <profile>
-      <id>ossrh</id>
+      <id>central</id>
       <distributionManagement>
         <snapshotRepository>
-          <id>ossrh</id>
-          <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+          <id>basex-publish</id>
+          <url>https://central.sonatype.com/repository/maven-snapshots/</url>
         </snapshotRepository>
-        <repository>
-          <id>ossrh</id>
-          <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
-        </repository>
       </distributionManagement>
       <build>
         <plugins>
@@ -268,6 +264,7 @@
                   <goal>sign</goal>
                 </goals>
                 <configuration>
+                  <keyname>cg@basex.org</keyname>
                   <!-- This is necessary for gpg to not try to use the pinentry programs -->
                   <gpgArguments>
                     <arg>--pinentry-mode</arg>
@@ -293,6 +290,18 @@
                 </goals>
               </execution>
             </executions>
+          </plugin>
+
+          <plugin>
+            <groupId>org.sonatype.central</groupId>
+            <artifactId>central-publishing-maven-plugin</artifactId>
+            <version>0.10.0</version>
+            <extensions>true</extensions>
+            <configuration>
+              <publishingServerId>basex-publish</publishingServerId>
+              <autoPublish>true</autoPublish>
+              <waitUntil>published</waitUntil>
+            </configuration>
           </plugin>
         </plugins>
       </build>
@@ -397,7 +406,7 @@
           <version>3.3.1</version>
           <configuration>
             <tagNameFormat>@{project.version}</tagNameFormat>
-            <releaseProfiles>ossrh</releaseProfiles>
+            <releaseProfiles>central</releaseProfiles>
             <scmCommentPrefix>Release: </scmCommentPrefix>
           </configuration>
         </plugin>
@@ -463,18 +472,6 @@
             <linkXRef>false</linkXRef>
             <consoleOutput>true</consoleOutput>
             <violationSeverity>warning</violationSeverity>
-          </configuration>
-        </plugin>
-
-        <plugin>
-          <groupId>org.sonatype.plugins</groupId>
-          <artifactId>nexus-staging-maven-plugin</artifactId>
-          <version>1.7.0</version>
-          <extensions>true</extensions>
-          <configuration>
-            <serverId>ossrh</serverId>
-            <nexusUrl>https://oss.sonatype.org/</nexusUrl>
-            <autoReleaseAfterClose>true</autoReleaseAfterClose>
           </configuration>
         </plugin>
 

